app/controllers/claims_controller.rb

```diff
@@ -38,6 +38,8 @@ def show
       session[:claim_postcode] = nil
       session[:claim_address_line_1] = nil
       redirect_to claim_path(current_policy_routing_name, "postcode-search") and return
+    elsif params[:slug] == "student-loan-amount" && params[:policy] == "student-loans"
+      ClaimStudentLoanDetailsUpdater.call(current_claim)
```

Comment:
@slorek valid point

This is a first pass at it, using the session (on a separate branch for now): https://github.com/DFE-Digital/claim-additional-payments-for-teaching/commit/901858433ac04510a523f9fc05fc3228c4bd5cef

The issue is that we'd still need to kick the user out of the journey when we detect that the total repayment is zero without waiting for user confirmation on a form (i.e. clicking a button). Otherwise I can try to discuss this with product. I am open to suggestions to address this scenario 🤔